.github/workflows/test.yaml

```diff
@@ -50,7 +50,8 @@ jobs:
     - name: golangci-lint
       uses: golangci/golangci-lint-action@v3
       with: # TODO: remove this when the deprecated function will be removed (issue #85)
-        args: --exclude SA1019 --timeout=5m -E ginkgolinter
+        args: --timeout=5m -v
+        version: v1.44.0
```

Comment:
The latest version of the golangci-lint is `v1.54.1` - any reason not to use it?

Reply:
As stated in the previous comment section, the idea was to install the `golangci-lint` binary using `go install` to avoid interferences from global packages.

However, the latest releases are requiring Go +v1.19 and the `go.mod` is referencing 1.18. We can decouple from the local development environment from the build one, it should be then well documented without giving anything for granted.

Reply:
1.18 is deprecated anyway. Maybe we better upgrade to 1.20 or 1.21?

Reply:
Alternatively, we can download the binary

Reply:
So @prometherion, I'm working on a new PR to bump golang, here: #257

It's now failing on the unit test, but this is as result of changes in #255. I'll try to fix it next week. But anyway, I think we need to find ways to use newer version of the golangci-lint. v1.44 is way too old.

Reply:
I plan to look at it next week. But if you'll have some time, here is the unit test failure log:
```
------------------------------
•! [PANICKED] [0.001 seconds]
Reconcile
/home/nunnatsa/GIT/cluster-api-provider-kubevirt/controllers/kubevirtcluster_controller_test.go:37
  reconcile cluster with finalizer and deletion time stamp
  /home/nunnatsa/GIT/cluster-api-provider-kubevirt/controllers/kubevirtcluster_controller_test.go:142
    [It] should succeed with the kubevirt cluster being deleted.
    /home/nunnatsa/GIT/cluster-api-provider-kubevirt/controllers/kubevirtcluster_controller_test.go:154

  Test Panicked
  In [It] at: /usr/local/go1.20.3/src/runtime/panic.go:260

  runtime error: invalid memory address or nil pointer dereference

  Full Stack Trace
    sigs.k8s.io/cluster-api-provider-kubevirt/controllers.(*KubevirtClusterReconciler).deleteExtraGVK(0x37b3ce0, 0xc000707100, {{0x2614d0d, 0x0}, {0x2614d0e, 0x2}, {0x2622de1, 0xd}})
        /home/nunnatsa/GIT/cluster-api-provider-kubevirt/controllers/kubevirtcluster_controller.go:292 +0x323
    sigs.k8s.io/cluster-api-provider-kubevirt/controllers.(*KubevirtClusterReconciler).reconcileDelete(0x37b3ce0, 0xc000707100, 0x7f37fced24a8?)
        /home/nunnatsa/GIT/cluster-api-provider-kubevirt/controllers/kubevirtcluster_controller.go:255 +0x5ec
    sigs.k8s.io/cluster-api-provider-kubevirt/controllers.(*KubevirtClusterReconciler).Reconcile(0x37b3ce0, {0x28db298?, 0xc00004c120}, {{{0x0?, 0x37e4790?}, {0x2631233?, 0x37e4790?}}})
        /home/nunnatsa/GIT/cluster-api-provider-kubevirt/controllers/kubevirtcluster_controller.go:155 +0xaaa
    sigs.k8s.io/cluster-api-provider-kubevirt/controllers_test.glob..func1.5.3()
        /home/nunnatsa/GIT/cluster-api-provider-kubevirt/controllers/kubevirtcluster_controller_test.go:167 +0x1ea
------------------------------
```

What we'll need to do is to mock the new `KubevirtClusterReconciler.APIReader`.